java/src/main/java/com/nedap/retail/example/Application.java

```diff
@@ -32,21 +29,30 @@
 
     private static RenosWebSocketClient client;
     private static ApiCaller api;
+    private static String BASE_URL;
 
     public static void main(final String[] args) {
         if (args.length == 0) {
             LOG.info("Please use URL of device as parameter, for example: http://localhost:8081");
             System.exit(0);
+        } else {
+            BASE_URL = trimTrailingSlash(args[0]);
         }
 
         final String baseUrl = trimTrailingSlash(args[0]);
 
-        api = new ApiCaller(baseUrl);
+        api = new ApiCaller(BASE_URL);
 
         LOG.info("Application starting...");
-        client = new RenosWebSocketClient(baseUrl);
+        client = new RenosWebSocketClient(BASE_URL);
 
         try (BufferedReader inputBuffer = new BufferedReader(new InputStreamReader(System.in))) {
+
+            LOG.info("Put in token:");
+            String token = inputBuffer.readLine();
```

Comment:
I noticed this version of the api example code isn't usable with the `pre-token` era. Is that a problem?